mono/mini/mini-runtime.c

```diff
@@ -1905,6 +1905,97 @@ mono_jit_map_is_enabled (void)
 
 #endif
 
+#if ENABLE_JIT_DUMP
+static FILE* perf_dump_file;
+uint64_t codeIndex = 0;
```

Comment:
`codeIndex` (renamed as `code_index`) can be a static variable inside `mono_emit_jit_dump()`, and be incremented inside the critical section protected by the mutex.